cmake/ccache.cmake

```diff
@@ -25,25 +25,27 @@ endif()
 execute_process(COMMAND ${CCACHE_EXECUTABLE} "-V" OUTPUT_VARIABLE CCACHE_VERSION)
 string(REGEX REPLACE "ccache version ([0-9\\.]+).*" "\\1" CCACHE_VERSION ${CCACHE_VERSION})
 
-if (CCACHE_VERSION VERSION_GREATER "3.2.0" OR NOT COMPILER_CLANG)
-    message(STATUS "Using ccache: ${CCACHE_EXECUTABLE} (version ${CCACHE_VERSION})")
-    set(LAUNCHER ${CCACHE_EXECUTABLE})
-
-   # debian (debhelpers) set SOURCE_DATE_EPOCH environment variable, that is
-   # filled from the debian/changelog or current time.
-   #
-   # - 4.0+ ccache always includes this environment variable into the hash
-   #   of the manifest, which do not allow to use previous cache,
-   # - 4.2+ ccache ignores SOURCE_DATE_EPOCH for every file w/o __DATE__/__TIME__
-   #
-   # Exclude SOURCE_DATE_EPOCH env for ccache versions between [4.0, 4.2).
-   if (CCACHE_VERSION VERSION_GREATER_EQUAL "4.0" AND CCACHE_VERSION VERSION_LESS "4.2")
-      message(STATUS "Ignore SOURCE_DATE_EPOCH for ccache")
-      set(LAUNCHER env -u SOURCE_DATE_EPOCH ${CCACHE_EXECUTABLE})
-   endif()
-
-   set (CMAKE_CXX_COMPILER_LAUNCHER ${LAUNCHER} ${CMAKE_CXX_COMPILER_LAUNCHER})
-   set (CMAKE_C_COMPILER_LAUNCHER ${LAUNCHER} ${CMAKE_C_COMPILER_LAUNCHER})
-else ()
-   message(${RECONFIGURE_MESSAGE_LEVEL} "Using ccache: No. Found ${CCACHE_EXECUTABLE} (version ${CCACHE_VERSION}) but disabled because of bug: https://bugzilla.samba.org/show_bug.cgi?id=8118")
-endif ()
+set (CCACHE_MINIMUM_VERSION 3.3)
+
+if (CCACHE_VERSION VERSION_LESS_EQUAL ${CCACHE_MINIMUM_VERSION})
+    message(FATAL_ERROR "Using ccache: no (found ${CCACHE_EXECUTABLE} (version ${CCACHE_VERSION}), the minimum required version is ${CCACHE_MINIMUM_VERSION}")
+endif()
+
+message(STATUS "Using ccache: ${CCACHE_EXECUTABLE} (version ${CCACHE_VERSION})")
+set(LAUNCHER ${CCACHE_EXECUTABLE})
+
+# debian (debhelpers) set SOURCE_DATE_EPOCH environment variable, that is
```

Comment:
Just in case, we don't use debhelpers anymore.

Reply:
Clarified the comment.